x/lockup/keeper/msg_server.go

```diff
@@ -39,26 +39,26 @@ func (server msgServer) LockTokens(goCtx context.Context, msg *types.MsgLockToke
 
 	// check if there's an existing lock from the same owner with the same duration.
 	// If so, simply add tokens to the existing lock.
-	locks, err := server.keeper.AddToExistingLock(ctx, owner, msg.Coins[0], msg.Duration)
+	lock, err := server.keeper.AddToExistingLock(ctx, owner, msg.Coins[0], msg.Duration)
 	if err != nil {
 		return nil, err
 	}
 
 	// return the lock id of the existing lock when successfully added to the existing lock.
-	if len(locks) > 0 {
+	if lock.ID != 0 {
```

Comment:
Is the idea here that if the lock exists (ID > 0), then we can assume that it was successfully added to?

Reply:
yes correct!